### PR TITLE
Add autofix for F504 and F522

### PR DESCRIPTION
**Pull request by squiddy**
Refs: #1144

### CODE REVIEW

File: src/pyflakes/fixes.rs [hunk 1]
```diff
@@ -73,3 +76,97 @@ pub fn remove_unused_imports(
...
+            rpar: dict.rpar.clone(),
+            elements: dict
+                .elements
+                .iter()
+                .filter_map(|e| match e {
+                    DictElement::Simple {
+                        key: Expression::SimpleString(name),
+                        ..
+                    } if unused_arguments
+                        .contains(&name.value.trim_matches(|c| c == '"' || c == '\'')) =>
```
Comment (charliermarsh) [inline src/pyflakes/fixes.rs:108]: What case is this handling?

Comment (squiddy) [reply src/pyflakes/fixes.rs:108]: This one is a bit weird, perhaps there is a better way. LibCST represents dictionary keys as SimpleStrings, so a key like `foo` is `"foo"`.

File: src/pyflakes/snapshots/ruff__pyflakes__tests__F504_F504.py.snap [hunk 2]
```diff
@@ -11,5 +11,12 @@ expression: checks
   end_location:
     row: 3
     column: 34
-  fix: ~
+  fix:
+    content: "{a: \"?\", }"
```
Comment (squiddy) [inline src/pyflakes/snapshots/ruff__pyflakes__tests__F504_F504.py.snap:15]: Note to myself, this is broken. Needs quoting.

File: src/pyflakes/fixes.rs (hunk 1 above)
Comment (charliermarsh) [reply src/pyflakes/fixes.rs:108]: You might be able to use `TRIPLE_QUOTE_PREFIXES` or `SINGLE_QUOTE_PREFIXES` from `src/docstrings/constants.rs`...?

Comment (squiddy) [reply src/pyflakes/fixes.rs]: I've added a `strip_quotes_and_prefixes` instead, because `SINGLE_QUOTE_PREFIXES` doesn't handle byte strings.

**Comment by charliermarsh** [comment]
Thanks for this, awesome work.

**Pull request merged**